controllers/grpcroute_controller.go

```diff
@@ -0,0 +1,376 @@
+		glog.V(6).Infof("Found aws-vpc-lattice for GRPCRoute for %v\n", grpcRoute.Spec())
+
+		return true
+	} else {
+		glog.V(6).Infof("Ignore non aws-vpc-lattice GRPCRoute !!! %v\n", grpcRoute.Spec())
+		return false
+	}
+}
+
+func (r *GRPCRouteReconciler) buildAndDeployModel(ctx context.Context, route core.Route) (core.Stack, *latticemodel.Service, error) {
+	httpLog := log.FromContext(ctx)
+
+	stack, latticeService, err := r.modelBuilder.Build(ctx, route)
+
+	if err != nil {
+
+		r.eventRecorder.Event(route.K8sObject(), corev1.EventTypeWarning,
+			k8s.GRPCRouteEventReasonFailedBuildModel, fmt.Sprintf("Failed build model due to %+v", err))
+		glog.V(6).Infof("buildAndDeployModel, Failed build model for %v due to %+v\n", route.Name(), err)
+
+		// Build failed
+		// TODO continue deploy to trigger reconcile of stale GRPCRoute and policy
+		return nil, nil, err
+	}
+
+	stackJSON, err := r.stackMashaller.Marshal(stack)
+	if err != nil {
+		//TODO
+		glog.V(6).Infof("error on r.stackMashaller.Marshal error %v \n", err)
+	}
+
+	httpLog.Info("Successfully built model:", stackJSON, "")
+
+	if err := r.stackDeployer.Deploy(ctx, stack); err != nil {
+		glog.V(6).Infof("GRPCRouteReconciler: Failed deploy %s due to err %v \n", route.Name(), err)
+
+		var retryErr = errors.New(lattice.LATTICE_RETRY)
+
+		if errors.As(err, &retryErr) {
+			r.eventRecorder.Event(route.K8sObject(), corev1.EventTypeNormal,
+				k8s.GRPCRouteEventReasonRetryReconcile, "retry reconcile...")
+
+		} else {
+			r.eventRecorder.Event(route.K8sObject(), corev1.EventTypeWarning,
+				k8s.GRPCRouteEventReasonFailedDeployModel, fmt.Sprintf("Failed deploy model due to %v", err))
+		}
+		return nil, nil, err
+	}
+
+	httpLog.Info("Successfully deployed model")
+
+	return stack, latticeService, err
+}
+
+func (r *GRPCRouteReconciler) reconcileGRPCRouteResource(ctx context.Context, grpcRoute *core.GRPCRoute) error {
+	glog.V(6).Infof("Beginning -- reconcileGRPCRouteResource, [%v]\n", grpcRoute)
+
+	if err := r.finalizerManager.AddFinalizers(ctx, grpcRoute.K8sObject(), grpcRouteFinalizer); err != nil {
+		r.eventRecorder.Event(grpcRoute.K8sObject(), corev1.EventTypeWarning, k8s.GRPCRouteEventReasonFailedAddFinalizer, fmt.Sprintf("Failed add finalizer due to %v", err))
+	}
+
+	_, _, err := r.buildAndDeployModel(ctx, grpcRoute)
+
+	//TODO add metric
+
+	if err == nil {
+		r.eventRecorder.Event(grpcRoute.K8sObject(), corev1.EventTypeNormal,
+			k8s.GRPCRouteEventReasonDeploySucceed, "Adding/Updating reconcile Done!")
+
+		serviceStatus, err1 := r.latticeDataStore.GetLatticeService(grpcRoute.Name(), grpcRoute.Namespace())
+
+		if err1 == nil {
+			r.updateGRPCRouteStatus(ctx, serviceStatus.DNS, grpcRoute)
+		}
+	}
+
+	return err
+
+}
+
+func (r *GRPCRouteReconciler) updateGRPCRouteStatus(ctx context.Context, dns string, coreRoute *core.GRPCRoute) error {
+	glog.V(6).Infof("updateGRPCRouteStatus: grpcRoute %v, dns %v\n", coreRoute, dns)
```

Comment:
What we cannot combine  and updateHTTPRouteStatus() and updateGRPCRouteStatus()  and use coreRoute core.Route as the last parameter? Or do you plan to combine it later?

Reply:
Yup! Done next revision